app/uk/gov/hmrc/individualsifapistub/domain/Income.scala

```diff
@@ -16,6 +16,142 @@
+  val weeklyPeriodNumberPattern = "^([1-9]|[1-4][0-9]|5[0-46])$".r
+
+  val payFrequencyValues = Seq("W1", "W2", "W4", "M1", "M3", "M6", "MA", "IO", "IR")
+
+  def isInPayFrequency(implicit rds: Reads[String]) : Reads[String] = {
+    verifying(value => payFrequencyValues.contains(value))
+  }
+}
+
+case class SaIncome(
+                     selfAssessment: Option[Double],
+                     allEmployments: Option[Double],
+                     ukInterest: Option[Double],
+                     foreignDivs: Option[Double],
+                     ukDivsAndInterest: Option[Double],
+                     partnerships: Option[Double],
+                     pensions: Option[Double],
+                     selfEmployment: Option[Double],
+                     trusts: Option[Double],
+                     ukProperty: Option[Double],
+                     foreign: Option[Double],
+                     lifePolicies: Option[Double],
+                     shares: Option[Double],
+                     other: Option[Double]
+                   )
+
+case class SaReturnType(
+                         utr: Option[String],
+                         caseStartDate: Option[String],
+                         receivedDate: Option[String],
+                         businessDescription: Option[String],
+                         telephoneNumber: Option[String],
+                         busStartDate: Option[String],
+                         busEndDate: Option[String],
+                         totalTaxPaid: Option[Double],
+                         totalNIC: Option[Double],
+                         turnover: Option[Double],
+                         otherBusinessIncome: Option[Double],
+                         tradingIncomeAllowance: Option[Double],
+                         address: Option[Address],
+                         income: Option[SaIncome]
+                       )
+
+case class SaTaxYearEntry(taxYear: Option[String], income: Option[Double], returnList: Option[Seq[SaReturnType]])
+
+object SaResponseObject {
+
+  val minValue = -9999999999.99
+  val maxValue = 9999999999.99
+
+  val utrPattern = "^[0-9]{10}$".r
+  val dateStringPattern = "^(((19|20)([2468][048]|[13579][26]|0[48])|2000)[-]02[-]29|((19|20)[0-9]{2}[-](0[469]|11)[-](0[1-9]|1[0-9]|2[0-9]|30)|(19|20)[0-9]{2}[-](0[13578]|1[02])[-](0[1-9]|[12][0-9]|3[01])|(19|20)[0-9]{2}[-]02[-](0[1-9]|1[0-9]|2[0-8])))$".r
+  val taxYearPattern = "^20[0-9]{2}$".r
+
+  def isMultipleOfPointZeroOne(value: Double): Boolean = (BigDecimal(value) * 100.0) % 1 == 0
+
+  def isInRange(value: Double): Boolean = value > minValue && value < maxValue
+
+  def paymentAmountValidator(implicit rds: Reads[Double]): Reads[Double] =
+    verifying[Double](value => isInRange(value) && isMultipleOfPointZeroOne(value))
+}
+
+case class IncomeSaRecord(id:Id, incomeSaResponse: IncomeSaResponse)
+
+case class IncomePayeRecord(id:Id, incomePayeResponse: IncomePayeResponse)
+
+case class IncomeSaResponse(sa: Option[Seq[SaTaxYearEntry]])
```

Comment:
Remove "Response" from class names (for consistency with 313, also because it looks weird when the user posts a response to the "create" methods in controller!) - rename the associated objects too